src/lib/createAuthContext.js

```diff
@@ -39,8 +40,19 @@ export default ({
   }
 
   const useToken = () => {
-    const { token } = useContext(context)
-    if (!token) {
+    const { token, setToken } = useContext(context)
+    if (token) {
+      const now = new Date()
+      const elapsed = now.getTime() - new Date(token.expires_at).getTime()
+      const slack = 1000
+      if(elapsed > slack){ 
```

Comment:
`useToken()` is a React hook, which means it gets called on `render()`.

This further means that if we then (at render time) detect that the token is expired, we refresh it. That's basically the right logic, but: refreshing the token is asynchronous, meaning that until the refresh is finished, we will return (in line 46) an expired token.

The contract with the users of this package is, however: `useToken()` either gives you a valid (non-expired) token, or no token at all. The README.md doesn't say this explicitly, but implicitly:

> ... This will ensure the user gets authenticated, before 
> anything wrapped by `Authenticated` gets mounted / rendered. ...

("Authenticated" should mean that there is a valid access token, not an expired one.)

Thus, the implementation here is not good, as it breaks this contract.

The right logic, IMHO, is: right after receiving a valid (and not-yet-expired) token, we set a timer that fires `slack` milliseconds before the token will expire. When the timer fires, we refresh the token, most likely (hopefully) before the (old) token expires. (`slack` currently is 1000ms, I believe it should be a little longer, maybe 10 seconds.)